src/clj/puppetlabs/services/jruby/jruby_agents.clj

```diff
@@ -85,13 +93,14 @@
    old-pool-state :- jruby-schemas/PoolState
    new-pool-state :- jruby-schemas/PoolState
    refill? :- schema/Bool]
-  (let [{:keys [config pool-state]} pool-context
+  (let [{:keys [config]} pool-context
+        pool-state-atom (get-in pool-context [:internal :pool-state])
```

Comment:
Would it make sense to call `(jruby-internal/get-pool-state pool-context)` here instead?